app/src/main/rust/src/lib.rs

```diff
@@ -71,44 +64,39 @@ where
 
 fn parse_marshal_inplace<F, R>(env: &mut JNIEnv, str: JByteBuffer, limit: jint, mut f: F) -> i32
 where
-    F: FnMut(&VDom, &Parser, &str) -> Option<R> + UnwindSafe,
+    F: FnMut(&VDom, &Parser, &str) -> Result<R, &'static str>,
     R: Serialize,
 {
     let ptr = env.get_direct_buffer_address(&str).unwrap();
     let html = unsafe {
         let buff = slice_from_raw_parts(ptr, limit as usize);
         from_utf8_unchecked(&*buff)
     };
-    let f = move || {
-        let dom = tl::parse(html, tl::ParserOptions::default()).unwrap();
+    let mut f = || {
+        let dom = tl::parse(html, tl::ParserOptions::default()).map_err(|_| "Illegal html!")?;
```

Comment:
If it's not html, the body usually contains error messages (e.g. temporary IP ban).
We need to pass that.

Reply:
Done